client/state/partner-portal/licenses/hooks/use-assign-license-mutation.ts

```diff
@@ -0,0 +1,28 @@
+import { useMutation, UseMutationOptions, UseMutationResult } from 'react-query';
+import { wpcomJetpackLicensing as wpcomJpl } from 'calypso/lib/wp';
+import { APILicenseAssign } from 'calypso/state/partner-portal/types';
+
+interface MutationAssignLicenseVariables {
+	licenseKey: string;
+	selectedSite: any;
```

Comment:
Could we increase the specificity of `selectedSite` by defining it as a number? AFAIK it should always be a blog ID

Reply:
I think there are cases where `selectedSite` is a site slug, no?

Reply:
>I think there are cases where selectedSite is a site slug, no?

The endpoint we call [will accept a site slug, full url or blog id](https://github.com/Automattic/jetpack-licensing-api/blob/master/spec.yml#L151-L166).
IMO the site url arguments are mostly for partners who call the API directly and aren't aware of blog ids (we cannot expect partners to be aware of blog ids, so we support multiple types of arguments).

> AFAIK it should always be a blog ID

The way this PR uses the mutation, we should always receive a blog ID.
Theoretically, the blog id is the most specific site identifier we have (think e.g. IDC) and since we're always aware of the blog ID in the licensing portal, I feel like we should enforce that.
It's an opinionated argument, so we can **_definitely_** leave it **_as-is_** since it works, but if we forcing it to be a number, it could help catch errors in the code whereas the current interface will allow anything (strings, numbers, objects, bools, null, etc) and then the endpoint will have to tell us if it looks OK 😄

Reply:
Fair enough. Updated in https://github.com/Automattic/wp-calypso/pull/60903/commits/c71fdab1a09f06f70099c950ac8b882a0a32db14!